src/main/java/hudson/remoting/RemoteClassLoader.java

```diff
@@ -281,6 +319,23 @@ along with the reference to the initiating ClassLoader (if the initiating ClassL
         }
     }
 
+    private void invokeClassLoadTestingHookIfNeeded() throws InterruptedException {
+        // Testing support only.
+        if (TESTING_CLASS_LOAD != null) {
+            TESTING_CLASS_LOAD.run();
+            if (Thread.currentThread().isInterrupted()) {
+                // Otherwise the interrupt isn't recognized and the test doesn't work.
+                throw new InterruptedException("loading was interrupted.");
+            }
```

Comment:
nit: maybe use a different runnable hook, which would directly throw the exception you need for your test, instead of "hardcoding" the exception here?
We can imagine test cases which would involve setting the interrupted flag here, but without expecting this `InterruptedException` (for instance something which would try to get a `RemotingException` from the `proxy.fetch` call in `loadRemoteClass`).